src/index/generateTrees/shared/findSharedParent.ts

```diff
@@ -2,16 +2,19 @@ import path from "path";
 
 /** Find the common parent directory between all paths. */
 export const findSharedParent = (paths: string[]) => {
-  if (paths.length === 1) return path.dirname(paths[0]);
+  const [firstPath] = paths;
+  if (paths.length === 1) return path.dirname(firstPath);
 
-  const parentPaths: string[] = [];
-  const parts: string[][] = paths.map(x => x.split("/"));
-  const firstParts = parts[0];
+  const [shortest, secondShortest] = paths.sort((a, b) => a.length - b.length);
 
-  firstParts.forEach((part, idx) => {
-    const allPartsMatch = parts.every(p => p[idx] === part);
+  const parentPaths: string[] = [];
+  const [shortestParts, secondShortestParts] = [
+    shortest,
+    secondShortest,
+  ].map(x => x.split("/"));
 
-    if (allPartsMatch) {
+  shortestParts.forEach((part, idx) => {
```

Comment:
I'd prefer to return a filter here instead. That makes quite a bit more clear and readable.